src/Collection.js

```diff
@@ -264,7 +264,7 @@ export default Component.extend({
         // Here, the currently visted record does not correspond to any record
         // in the collection, so (optionally) instantiate this record and insert
         // it into the collection
-        record = this.mapper ? this.mapper.createRecord(record, opts) : record
+        record = this.mapper && !this.mapper.is(record) ? this.mapper.createRecord(record, opts) : record
```

Comment:
This would be a breaking change, as `createRecord` does more than just wrap the props in the Mapper's Record class. It also checks the record's relations and makes sure have been wrapped too. `createRecord` doesn't double wrap records, so you don't need the `!this.mapper.is(record)` check.